editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/wizards/pages/FuseIntegrationProjectWizardTemplatePage.java

```diff
@@ -195,7 +198,53 @@ public void refresh() {
 		if (listTemplates != null) {
 			listTemplates.getViewer().refresh();
 			listTemplates.getViewer().expandAll();
+			selectDefaultTemplates(listTemplates);
+		}
+	}
+
+	private void selectDefaultTemplates(FilteredTree listTemplates) {
+		TreeViewer viewer = listTemplates.getViewer();
+		Set<TreeItem> unfilteredItems = retrieveAllTreeItems(viewer);
+		if (viewer.getSelection().isEmpty()) {
+			if (unfilteredItems.size() == 1) {
+				viewer.setSelection(new StructuredSelection(unfilteredItems.iterator().next().getData()));
+			} else {
+				TemplateItem defaultToSelect = determineDefaultTemplateToSelect(unfilteredItems, environment);
+				if (defaultToSelect != null) {
+					viewer.setSelection(new StructuredSelection(defaultToSelect));
+				}
```

Comment:
and if there is no default and more items in list we do not select anything?

Reply:
there is always a default, there is an automated test for that.
Even if there is no default, I think that the best is to not select anything, it would mean that we don't know what is the recommended template, so seems fair to force user to go choose one explictily